subsys/dfu/src/dfu_context_handler.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <logging/log.h>
+#include <dfu/mcuboot.h>
+#include <mcuboot_dfu_ctx.h>
+#ifdef CONFIG_DFU_CTX_MODEM_UPDATE_SUPPORT
+#include <modem_dfu_ctx.h>
+#endif
+#include <dfu/dfu_context_handler.h>
+#include "dfu_context_handler_internal.h"
+
+#define MCUBOOT_IMAGE 1
+#define MODEM_DELTA_IMAGE 2
+
+LOG_MODULE_REGISTER(dfu_context_handler, CONFIG_DFU_CTX_LOG_LEVEL);
+static struct dfu_ctx *ctx;
+
+static inline void log_img_hdr(struct mcuboot_img_header_v2 *img_hdr)
+{
+	struct mcuboot_img_sem_ver ver = img_hdr->sem_ver;
+
+	LOG_DBG("magic:\t\t0x%x\n", img_hdr->magic);
+	LOG_DBG("load address:\t0x%x\n", img_hdr->load_addr);
+	LOG_DBG("header size:\t%d\n", img_hdr->hdr_size);
+	LOG_DBG("pad1:\t\t%d\n", img_hdr->_pad1);
+	LOG_DBG("image_size:\t%d\n", img_hdr->img_size);
+	LOG_DBG("flags:\t0x%x\n", img_hdr->flags);
+	LOG_DBG("image version:\t%d.%d.%d.%d\n", ver.major, ver.minor,
+		ver.revision, ver.build_num);
+}
+
+int dfu_ctx_init(int img_type)
+{
+	if (img_type == MCUBOOT_IMAGE) {
+		ctx = &mcuboot_dfu_ctx;
+	}
+#ifdef CONFIG_DFU_CTX_MODEM_UPDATE_SUPPORT
+	if (img_type == MODEM_DELTA_IMAGE) {
+		ctx = &modem_dfu_ctx;
+	}
+#endif /* CONFIG_DFU_CTX_LOG_LEVEL */
+	if (ctx != NULL) {
+		return ctx->init();
+	}
+	LOG_ERR("Unknown image type");
+	return -ENOTSUP;
+}
+
+
+int dfu_ctx_write(u8_t *buf, size_t len)
```

Comment:
See if you can work with `void *` so the caller doesn't have to cast for no reason :)

Reply:
Sure will fix it :+1: